File: Data Structures & Algorithms/majority-element/submission-2.py
'''from collections import Counter

class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hash_table= Counter(nums)
        for key, value in hash_table.items():
            if value == max(hash_table.values()):
                return key'''

class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        freq = dict()
        for n in nums:
            freq[n] = freq.get(n,0) +1
            if freq[n] > len(nums)/2:
                return n
        

File: Data Structures & Algorithms/sort-colors/submission-2.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        unordered_map<int, int> freq;

        for (int n : nums) {
            freq[n]++;
        }

        vector<int> result;

        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < freq[i]; j++) {
                result.push_back(i);
            }
        }

        nums = result;
    }
};